cmd/cli/sign.go

```diff
@@ -24,12 +28,32 @@ func Sign() cli.Command {
 
 			path = path + "/" + name
 
-			priv, err := utils.Read(path, "key.bin")
+			pub, priv, err := keys.GenerateDH()
+			if err != nil {
+				return errors.Wrap(err, "can not generate keys")
+			}
+			err = utils.Write(path, "pub.bin", pub[:])
+			if err != nil {
+				return errors.Wrap(err, "can not write in pub.bin")
+			}
+			err = api.Publish(name, pub[:])
```

Comment:
Nope, you didn't need publish any keys in sign command. `publish` ans `sign` - it's different commands, where output of `publish` can be used in the `sign` command